UPDATING.md

```diff
@@ -0,0 +1,41 @@
+# Upgrade / Migration Guide
+
+## Version 1.x to 2.x
+
+### Modularization
+
+The SDK has now been divided into three modules: `lib`, `java` and `android`. The `lib` module contains code that is common to both `java` and `android`.
+This is better than the previous approach whereby `java` and `android` were monoliths linked only by shared source code sets.
+
```

Comment:
IMO it might be a bit too much detail but this informs why suddenly we have 3 ably-java related dependencies in Maven Central and even might be an inspiration to other developers who are in a similar situation with their projects :wink: So I'd say let's leave it, but I'd also love to hear @QuintinWillison opinion on this :thinking:

Reply:
I rather thought from the point of a developer who is upgrading their library. This additional information can be a bit overwhelming when they try to upgrade dependency on their project. I would recommend that if you think there is a need to provide this, then do this in the end. Maybe put a title like "Reasons why you had to to all these changes" and explain the reasoning. So it becomes optional for people who wants to read it.

Reply:
I would support moving it to the end, but I do think it is worth keeping. I would also suggest that an additional paragraph could be added that explains that this change introduces a new transitive dependency but does not otherwise change the way that the Ably SDK dependency is imported into app projects.

Reply:
I've moved this section and added a paragraph about the dependency :wink: https://github.com/ably/ably-java/pull/768/commits/7cff2936fdd7a589ac4d693ce092a83b2243dcdc